src/main/java/configurationslicing/AbstractJob.java

```diff
@@ -0,0 +1,87 @@
+package configurationslicing;
+
+import java.io.IOException;
+
+import hudson.model.AbstractProject;
+import hudson.model.Job;
+import hudson.triggers.Trigger;
+import hudson.triggers.TriggerDescriptor;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+
+/**
+ * Object that tries to be missing link between {@link AbstractProject}
+ * and {@link WorkflowJob} for the purpose of this plugin.
+ *
+ * @author Michal Slusarczyk
+ */
+public class AbstractJob {
+
+    private Job item;
+
+    public AbstractJob(Job item) {
+        this.item = item;
+    }
+
+    public static AbstractJob fix(Job item) {
+        return new AbstractJob(item);
+    }
+
+    public boolean isConcurrentBuilds() {
+        if (item instanceof AbstractProject)
+            return ((AbstractProject) item).isConcurrentBuild();
+        if (item instanceof WorkflowJob)
+            return ((WorkflowJob) item).isConcurrentBuild();
+
+        return true;
+    }
+
+    public void makeConcurrentBuilds(boolean value) throws IOException {
+        if (item instanceof AbstractProject)
+            ((AbstractProject) item).setConcurrentBuild(value);
+        if (item instanceof WorkflowJob)
+            ((WorkflowJob) item).setConcurrentBuild(value);
+
+        throw new IOException("Unsupported job type");
+    }
+
+    public <T extends Trigger> T getTrigger(Class<T> clazz) {
+        if (item instanceof AbstractProject)
+            return (T) ((AbstractProject) item).getTrigger(clazz);
+        if (item instanceof WorkflowJob) {
+            return getTrigger((WorkflowJob) item, clazz);
```

Comment:
Some of these things are better handled via `ParameterizedJob` by the way.

Reply:
or `SCMTriggerItem`

Reply:
@jglick  
Hey new and kinda trying to make sense of the codebase and Jenkins Core API.

Can you be more specific? What would ``SCMTriggerItem`` and ``ParameterizedJob`` replace?

Reply:
Explicit checks for `AbstractProject` and `WorkflowJob`. Not sure all functionality you are looking for here would be covered.